File: src/py-opentimelineio/opentimelineio/schema/effect.py
# SPDX-License-Identifier: Apache-2.0
# Copyright Contributors to the OpenTimelineIO project

from .. core._core_utils import add_method
from .. import _otio


@add_method(_otio.Effect)
def __str__(self):
    return (
        "Effect("
        "{}, "
        "{}, "
        "{}, "
        "{}"
        ")".format(
            str(self.name),
            str(self.effect_name),
            str(self.metadata),
            str(self.enabled)
        )
    )


@add_method(_otio.Effect)
def __repr__(self):
    return (
        "otio.schema.Effect("
        "name={}, "
        "effect_name={}, "
        "metadata={}, "
        "enabled={}"
        ")".format(
            repr(self.name),
            repr(self.effect_name),
            repr(self.metadata),
            repr(self.enabled)
        )
    )

File: src/opentimelineio/effect.cpp
// SPDX-License-Identifier: Apache-2.0
// Copyright Contributors to the OpenTimelineIO project

#include "opentimelineio/effect.h"
#include "opentimelineio/missingReference.h"

namespace opentimelineio { namespace OPENTIMELINEIO_VERSION {

Effect::Effect(
    std::string const&   name,
    std::string const&   effect_name,
    AnyDictionary const& metadata,
    bool enabled)
    : Parent(name, metadata)
    , _effect_name(effect_name)
    , _enabled(enabled)
{}

Effect::~Effect()
{}

bool
Effect::read_from(Reader& reader)
{
    return reader.read("effect_name", &_effect_name)
           && reader.read_if_present("enabled", &_enabled)
           && Parent::read_from(reader);
}

void
Effect::write_to(Writer& writer) const
{
    Parent::write_to(writer);
    writer.write("effect_name", _effect_name);
    writer.write("enabled", _enabled);
}

}} // namespace opentimelineio::OPENTIMELINEIO_VERSION
